matrix-e2e.c

```diff
@@ -47,6 +52,14 @@ typedef struct _MatrixE2ERoomData {
     GHashTable *megolm_sessions_inbound;
 } MatrixE2ERoomData;
 
+typedef struct _MatrixOlmSession {
```

Comment:
you can end up with quite a lot of these, so I'm not sure a linked list is the right structure. You should probably hash them by sender key.